### PR TITLE
RUBY-3104 remove tests < 3.6

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

File: lib/mongo/collection.rb [hunk 1]
```diff
@@ -372,13 +372,13 @@ def capped?
...
     def create(opts = {})
       # Passing read options to create command causes it to break.
-      # Filter the read options out.
+      # Filter the read options out. The same is true with session
+      # since it gets passed to the operation and fails BSON serialization.
       # TODO put the list of read options in a class-level constant when
       # we figure out what the full set of them is.
-      options = Hash[self.options.reject do |key, value|
-        %w(read read_preference read_concern).include?(key.to_s)
+      options = Hash[self.options.merge(opts).reject do |key, value|
```
Comment (p-mongo) [inline lib/mongo/collection.rb:379]: This is a behavior change and should be in its own PR & ticket.

**Review by p-mongo — DISMISSED** [review]
I apologize I read the PRs out of sequence.

This is on top of https://github.com/mongodb/mongo-ruby-driver/pull/2610 which does the behavior changes and the ticket is for behavior changes.

**Review by p-mongo — APPROVED** [review]

**Review by comandeo — APPROVED** [review]
